### ISSUE TITLE
The TSDAE-scidocs model from HuggingFace doesn't seem to work as expected.

### ISSUE THREAD

**Issue by calee88** [UKPLab/sentence-transformers#1617]
I'm basically trying to make a decode function for a SentenceTransformer model, which can translate a sentence embedding to the original sentence. Before doing that, I'm checking whether the model is working. Below is the code that I tried. I expected the output to be similar to the label, but it was totally different. What am I missing? Thank you for reading the question.
```python
from sentence_transformers import SentenceTransformer, losses, models, datasets
from torch.utils.data import DataLoader
from torch import Tensor

import torch


def batch_to_device(batch, target_device):
  """
    send a pytorch batch to a device (CPU/GPU)
    """
  for key in batch:
    if isinstance(batch[key], Tensor):
      batch[key] = batch[key].to(target_device)
  return batch


def main():
  dataset = 'scidocs'
  model_name_or_path = f'kwang2049/TSDAE-{dataset}'
  model = SentenceTransformer(model_name_or_path, device='cpu')
  model[1] = models.Pooling(
      model[0].get_word_embedding_dimension(),
      pooling_mode='cls')  # Note this model uses CLS-pooling

  sentences = [
      'Learning sentence embeddings often requires a large amount of labeled data.',
      'However for most tasks and domains, labeled data is seldom available and creating it is expensive.'
  ]

  # Create the special denoising dataset that adds noise on-the-fly
  train_dataset = datasets.DenoisingAutoEncoderDataset(sentences)

  # DataLoader to batch your data
  train_dataloader = DataLoader(train_dataset, batch_size=8, shuffle=True)
  train_dataloader.collate_fn = model.smart_batching_collate

  # Use the denoising auto-encoder loss
  train_loss = losses.DenoisingAutoEncoderLoss(model, tie_encoder_decoder=True)

  for data in train_dataloader:
    features, labels = data
    features = list(
        map(lambda batch: batch_to_device(batch, model._target_device),
            features))
    labels = labels.to(model._target_device)

    source_features, target_features = tuple(features)
    reps = train_loss.encoder(target_features)[
        'sentence_embedding']  # (bsz, hdim)

    # Prepare input and output
    target_length = target_features['input_ids'].shape[1]
    decoder_input_ids = target_features['input_ids'].clone(
    )[:, :target_length - 1]
    label_ids = target_features['input_ids'][:, 1:]

    # Decode
    decoder_outputs = train_loss.decoder(
        input_ids=decoder_input_ids,
        inputs_embeds=None,
        attention_mask=None,
        encoder_hidden_states=reps[:, None],  # (bsz, hdim) -> (bsz, 1, hdim)
        encoder_attention_mask=source_features['attention_mask'][:, 0:1],
        labels=None,
        return_dict=None,
        use_cache=False)

    lm_logits = decoder_outputs[0]

    print(lm_logits.max(-1)[1])
    print(label_ids)


if __name__ == '__main__':
  main()

```
output:
tensor([[ 3742,  8598,  3029, 12204,  4203, 25523,  9023,   101,  2691,  4263,
          4761, 19027,  5573,  4367, 10451, 23078, 22160,  2911,  4962],
        [ 3742, 13434,  2303, 20688,   101,  4642,  2918,  9499,  2030,  7386,
         14066,  8017, 16405, 10880,  2030,  3446, 18639, 17191,  4168]])
tensor([[ 4083,  6251,  7861,  8270,  4667,  2015,  2411,  5942,  1037,  2312,
          3815,  1997, 12599,  2951,  1012,   102,     0,     0,     0],
        [ 2174,  2005,  2087,  8518,  1998, 13100,  1010, 12599,  2951,  2003,
         15839,  2800,  1998,  4526,  2009,  2003,  6450,  1012,   102]])

**Comment by calee88** [comment]
I got it. The model was not a decoder, so it doesn't contain weights for crossattention. I think I need to train a model by myself. Closing the issue.
